Model/Data/AmwalOrderDetails.php

```diff
@@ -66,6 +67,22 @@ public function getOrderDetails($amwalOrderId)
         ];
     }
 
+
+    /**
+     * @param string $orderId
+     * @return array
+     */
+    public function getOrderByOrderId($orderId)
+    {
+        // Get order by order ID
+        $order = $this->getOrderById($orderId);
+        $order->setData('order_url', $this->getOrderUrl($order));
+
+        return [
+            'order' => $order->getData(),
```

Comment:
If you expect a "flat" array without objects I think you can also use `toArray` here (but do first verify if this works for your use case)
```suggestion
            'order' => $order->toArray(),
```